dora/core/client/fs/src/main/java/alluxio/client/file/DoraCacheFileSystem.java

```diff
@@ -237,8 +237,8 @@ public PositionReader openPositionRead(URIStatus status, OpenFilePOptions option
             .setMountId(DUMMY_MOUNT_ID)
             .build();
     return mDoraClient.createNettyPositionReader(status, openUfsBlockOptions,
-        new CloseableSupplier<>(() ->
-            mDelegatedFileSystem.openPositionRead(status, mergedOptions)));
+        mUfsFallbackEnabled ? new CloseableSupplier<>(() ->
```

Comment:
might be out of scope of this pr, I might prefer to use Optional to avoid using null value

Reply:
changed to use `Optional`